constants.py

```diff
@@ -76,9 +80,13 @@ def _merge_vars(base_vars: JsonType, vars: JsonType) -> None:
 # NOTE: pyinstaller will set this to its own executable when building,
 # detect this to use __file__ and main.py redirection instead
 SELF_PATH = Path(sys.argv[0]).absolute()
+APPIMAGE_PATH = IS_APPIMAGE and Path(os.environ["APPIMAGE"]).absolute()
 if SELF_PATH.stem == "pyinstaller":
     SELF_PATH = Path(__file__).with_name("main.py").absolute()
```

Comment:
```suggestion
if IS_APPIMAGE:
    SELF_PATH = Path(os.environ["APPIMAGE"]).absolute()
else:
    # NOTE: pyinstaller will set sys.argv[0] to its own executable when building,
    # detect this to use __file__ and main.py redirection instead
    SELF_PATH = Path(sys.argv[0]).absolute()
    if SELF_PATH.stem == "pyinstaller":
        SELF_PATH = Path(__file__).with_name("main.py").absolute()
```

This is where the "reversal" of paths should be happening. If `os.environ["APPIMAGE"]` points to the executable, that's what should be used as the `SELF_PATH`.